src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs

```diff
@@ -111,8 +112,9 @@ public class MergeBlockDownloader : BlockDownloader
                     $"MergeBlockDownloader GetCurrentNumber: currentNumber {currentNumber}, beaconPivotExists: {_beaconPivot.BeaconPivotExists()}, BestSuggestedBody: {_blockTree.BestSuggestedBody?.Number}, BestKnownNumber: {_blockTree.BestKnownNumber}, BestPeer: {bestPeer}, BestKnownBeaconNumber {_blockTree.BestKnownBeaconNumber}");
 
             bool HasMoreToSync()
-                => currentNumber < _blockTree.BestKnownBeaconNumber &&
-                   bestPeer.HeadNumber > _blockTree.BestKnownNumber;
+                => (currentNumber < _blockTree.BestKnownBeaconNumber &&
+                    bestPeer.HeadNumber > _blockTree.BestKnownNumber)
+                   || _beaconPivot.BeaconPivotExists();
```

Comment:
Let's imagine a situation like this:

we have beaconPivot (X) and our CL started progressing. So we're know X+10. We don't need to download blocks from X+1, X+2, X+3 but unfortunately we need to download receipts. So I am not sure if the condition is the best

Reply:
Why do we need to download receipts?

Reply:
for CL clients and eth_getLogs

Reply:
I don't know either. The `currentNumber` is not used in `MergeBlockDownloader` but its here....

Reply:
How about this?
```
            bool HasMoreToSync(out BlockHeader[]? headers, out int headersToRequest)
            {
                if (_logger.IsDebug)
                    _logger.Debug($"Continue full sync with {bestPeer} (our best {_blockTree.BestKnownNumber})");

                headersToRequest = Math.Min(_syncBatchSize.Current, bestPeer.MaxHeadersPerRequest());
                if (_logger.IsTrace)
                    _logger.Trace(
                        $"Full sync request {currentNumber}+{headersToRequest} to peer {bestPeer} with {bestPeer.HeadNumber} blocks. Got {currentNumber} and asking for {headersToRequest} more.");
                headers = _chainLevelHelper.GetNextHeaders(headersToRequest);
                if (headers == null || headers.Length == 0)
                {
                    if (_logger.IsTrace)
                        _logger.Trace("Chain level helper got no headers suggestion");
                    return false;
                }

                return true;
            }

            while (HasMoreToSync(out BlockHeader[]? headers, out int headersToRequest))
            {
```

Reply:
Updated